include/maidsafe/crux/detail/multiplexer.hpp

```diff
@@ -313,7 +353,15 @@ inline void multiplexer::stop_receive()
 
     if (--receive_calls == 0)
     {
-        was_closed = true;
+        // The fact that we're here means that the last socket that
+        // was receiving just got closed. To prevent the io_service
+        // from blocking, we fulfill the last receive request by
+        // sending ourself an empty packet.
+        auto self = shared_from_this();
+        next_layer().async_send_to
```

Comment:
That remark pertains to old Windows versions. Does MaidSafe even support Windows XP?